README.md

```diff
@@ -65,6 +65,57 @@ widely used, while draft-03 is long since deprecated.
 If you see anything missing from the current supported drafts, or incorrect
 on any draft still accepting bug fixes, please file an issue or submit a PR.
 
+Optional Tests
+--------------
+
+Tests in the `optional/` directory test optional validation behavior. Validators
+only need to execute tests for implemented features.
+
+There are different sets of optional features for each draft. They are
+summarized below.
+
+1. Arbitrary-precision number support: JSON allows numbers of
+   arbitrary precision.\
+   See: [6. Numbers](https://tools.ietf.org/html/rfc8259#section-6)\
+   Also see:
+   * 2019-09: [4.2. Validation of Numeric Instances](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.4.2)
+   * 7: [4.2. Validation of Numeric Instances](https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-4.2)
+   * 6: [3.2. Validation of numeric instances](https://tools.ietf.org/html/draft-wright-json-schema-validation-01#section-3.2)
+   * 4: [3.2. Validation of numeric instances](https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-3.2)
+   * 3: No link
+2. Encoded content in a string: The `"contentMediaType"` property is optional.\
+   See:
+   * 2019-09: [8.4. contentMediaType](https://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.8.4)
+   * 7: [8.4. contentMediaType](https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-8.4)
+3. ECMA 262 regex quirks.\
+   See:
+   * 2019-09: [6.4. Regular Expressions](https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.6.4)
+   * 7: [4.3. Regular Expressions](https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-4.3)
+   * 6: [3.3. Regular expressions](https://tools.ietf.org/html/draft-wright-json-schema-validation-01#section-3.3)
+   * 4: [3.3. Regular expressions](https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-3.3)
+   * 3: https://tools.ietf.org/html/draft-zyp-json-schema-03 (look for the term
+     "regular expression")
+4. References to unknown keywords: These may optionally be processed.\
+   See:
+   * 2019-09: [6.5. Extending JSON Schema](https://json-schema.org/draft/2019-09/json-schema-core.html#rfc.section.6.5)
```

Comment:
These sections are about adding new validation keywords -- not exactly about what the subject of these tests were -- which aren't about using extended keywords for validation, they're about whether one can use a JSON `$ref` to reference an arbitrary part of the document (with no behavior, just as a dereference)

I *think* as I said in that PR that the language saying you don't need to support that is new in 2019-09 (But I don't think it's this section)

Reply:
Well, I don't fully understand this yet (See #349 and #348]. I'd love some help with the working of this section.